server/main.lua

```diff
@@ -304,54 +349,79 @@ RegisterNetEvent('qb-banking:updatePin', function(pin)
         local xPlayer = QBCore.Functions.GetPlayer(src)
         while xPlayer == nil do Wait(0) end
 
-        --   _char:Bank().UpdateDebitCardPin(pin)
-        TriggerClientEvent('qb-banking:openBankScreen', src)
-        TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.updated_pin'))
+      
+        local auth = validatePosition(GetEntityCoords(GetPlayerPed(src)))
+        if auth.isNearBank then
+              --   _char:Bank().UpdateDebitCardPin(pin)
+            TriggerClientEvent('qb-banking:openBankScreen', src)
+            TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.updated_pin'))
+        else
+            TriggerClientEvent('QBCore:Notify', src, Lang:t('error.too_far_away'), "error")
+            TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', "red", "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** has attempted to change a debit card pin whilst not at a banking location, Nearest bank = "..auth.nearestDistance)
+        end
     end
 end)
 
 RegisterNetEvent('qb-banking:savingsDeposit', function(amount)
     local src = source
     local xPlayer = QBCore.Functions.GetPlayer(src)
     while xPlayer == nil do Wait(0) end
-    local currentBank = xPlayer.Functions.GetMoney('bank')
-
-    if tonumber(amount) <= currentBank then
-        local bank = xPlayer.Functions.RemoveMoney('bank', tonumber(amount))
-        local savings = savingsAccounts[xPlayer.PlayerData.citizenid].AddMoney(tonumber(amount), Lang:t('info.current_to_savings'))
-        while bank == nil do Wait(0) end
-        while savings == nil do Wait(0) end
-        TriggerClientEvent('qb-banking:openBankScreen', src)
-        TriggerClientEvent('qb-banking:successAlert', src, Lang:t('success.savings_deposit', {value = tostring(amount)}))
-        TriggerEvent('qb-log:server:CreateLog', 'banking', 'Banking', 'lightgreen', "**"..GetPlayerName(xPlayer.PlayerData.source) .. " (citizenid: "..xPlayer.PlayerData.citizenid.." | id: "..xPlayer.PlayerData.source..")** made a savings deposit of $"..tostring(amount).." successfully..")
+    
+    local auth = validatePosition(GetEntityCoords(GetPlayerPed(src)))
+    if auth.isNearBank then
+        local currentBank = xPlayer.Functions.GetMoney('bank')
+        if tonumber(amount) <= currentBank then
+            local bank = xPlayer.Functions.RemoveMoney('bank', tonumber(amount))
+            local savings = savingsAccounts[xPlayer.PlayerData.citizenid].AddMoney(tonumber(amount), Lang:t('info.current_to_savings'))
+            while bank == nil do Wait(0) end
+            while savings == nil do Wait(0) end
```

Comment:
Please don't use while loops here as the variables are not getting assigned later anytime, if it is nil it stays nil, so please make a check like this `if not bank then return end` you can add a message to that check as well if you want